app/services/taxonomy/fetch_all_taxons.rb

```diff
@@ -1,5 +1,5 @@
 module Taxonomy
-  class TaxonFetcher
+  class FetchAllTaxons
```

Comment:
What if we went with that name and that API, but put this in `app/models`? All this class is doing is wrapping responses from our persistence layer, i.e. - the pub api.